src/context/shoppingItemContext/shoppingItemProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import { useMemo } from 'react';
+import useShoppingItemList from '../../hook/useShoppingItemList';
+import ShoppingItemContext from './shoppingItemContext';
+
+const ShoppingItemProvider = ({ children }: { children: React.ReactNode }) => {
+  const {
+    data,
+    error,
+    isLoading,
+    selectCategory,
+    selectSort,
+    category,
+    sortType,
+    retryFetch,
+  } = useShoppingItemList();
+
+  const value = useMemo(
+    () => ({
+      data,
+      error,
+      isLoading,
+      selectCategory,
+      selectSort,
+      category,
+      sortType,
+      retryFetch,
+    }),
+    [
+      data,
+      error,
+      isLoading,
+      selectCategory,
+      selectSort,
+      category,
+      sortType,
+      retryFetch,
+    ]
+  );
+
+  return (
+    <ShoppingItemContext.Provider value={value}>
+      {children}
+    </ShoppingItemContext.Provider>
+  );
+};
```

Comment:
```suggestion
const ShoppingItemProvider = ({ children }: { children: React.ReactNode }) => {
  const shoppingItemList = useShoppingItemList();

  return (
    <ShoppingItemContext.Provider value={shoppingItemList}>
      {children}
    </ShoppingItemContext.Provider>
  );
};
```
이렇게 표현할 수 있도록 해주면 어떨까요?

Reply:
babbab0e691bb2bf2d6b437b33ebd7f3d1d912be

수정했습니다!